docs/architecture/adr/0018-feature-management.md

```diff
@@ -0,0 +1,90 @@
+frequently with less development in isolation. Quality assurance teams desire to keep said quality
+of the products in place as do operations teams with respect to reliability and performance.
+Features can often be delivered in small chunks over time in parallel and there is a need to control
+their impact on systems by limiting their release to specific audiences and enabling
+experimentation.
+
+## Considered Options
+
+- **Add / change functionality directly** - Make changes via the SDLC and code review and merge into
+  the mainline release branch once approved. Work through issues and hotfixes as needed, alongside
+  other feature development.
+- **Implement a feature management system internally** - Utilize frameworks in code and store
+  feature flags and other components in our own storage, streaming their updates in a proprietary
+  way if at all (e.g. only loading configuration at application startup).
+- **Adopt a feature management system with local fallback** - Implement a service provider's
+  offering for more robust feature management capabilities such as real-time streaming and user /
+  context targeting. Support a local configuration for self-hosted installations that may want to
+  test or adopt a feature that isn't fully supported yet.
+
+## Decision Outcome
+
+Chosen option: **Adopt a feature management system with local fallback**.
+
+### Positive Consequences
+
+- Robust featureset for flags and their variations.
+- Protection from changes and targeted impact, along with a speed in overall delivery (with managed
+  functionality assumed to be turned "off").
+- Context-sensitive application of features.
+- Logging and traceability of who can experience or experiment with a feature.
+
+### Negative Consequences
+
+- Costs for selecting a service provider.
+
+### Plan
+
+The [server][server] codebase will adopt a .NET SDK for a service provider that offers feature
+management. Only the serverside SDK will be used at first to manage access and cost and feature
+states will be communicated down to calling clients where appropriate via API response elements. New
+features will be set up inside the service provider's platform, and changes to them will be streamed
+to the running applications. Access to the provider will be controlled internally.
+
+To facilitate when feature states need to be used by clients amongst other configuration, a new API
+will be developed to provide, amongst potentially other things, the server version and other build
+characteristics such as commit hash as well as a collection of configurations values. Some of these
+values are already maintained persistently and will be intermixed with feature keys. Clients will
+refresh configuration upon startup, when their local configuration is updated, and when sync events
+come in.
+
+Contexts -- generally representing individual users -- will be established that communicate to the
```

Comment:
This is a bit unclear, first we mention it being individual users, but a few lines below it's organizations and providers.

Reply:
I removed some confusing wording. While it _is_ about a user as the requestor, multiple contexts will be built. This allows more dynamic targeting to perhaps just one "kind" of context, or even multiple.